src/app/ui/question/question.component.ts

```diff
@@ -16,22 +19,54 @@ import {Course} from '../../entities/course';
 })
 export class QuestionComponent implements OnInit {
 
-  private id: string;
+  private qId: string;
   question$: Observable<QuestionId>;
   solutions$: Observable<SolutionId[]>;
   topics$: Observable<TopicWithCreatorId[]>;
   course$: Observable<Course>;
   isAdmin$: Observable<boolean>;
+  userId: string;
+  hasSolved = false;
+  solvedQuestion: SolvedQuestionId = null;
+  solvedQuestions$: Observable<SolvedQuestionId[]>;
 
-  constructor(private route: ActivatedRoute, private db: DbService, private auth: AuthService) {
-    this.id = this.route.snapshot.paramMap.get('id');
-    this.topics$ = this.db.getTopicsForQuestion(this.id);
-    this.question$ = this.db.getQuestion(this.id);
-    this.solutions$ = this.db.getSolutions(this.id);
-    this.isAdmin$ = this.db.getQuestion(this.id).pipe(flatMap(q => this.auth.isAdminForCourse(q.course)));
-    this.course$ = this.db.getQuestion(this.id).pipe(flatMap(q => this.db.getCourse(q.course)));
+  constructor(private route: ActivatedRoute, private db: DbService, private auth: AuthService, private snackBar: MatSnackBar) {
+    this.qId = this.route.snapshot.paramMap.get('id');
+    this.topics$ = this.db.getTopicsForQuestion(this.qId);
+    this.question$ = this.db.getQuestion(this.qId);
+    this.solutions$ = this.db.getSolutions(this.qId);
+    this.isAdmin$ = this.db.getQuestion(this.qId).pipe(flatMap(q => this.auth.isAdminForCourse(q.course)));
+    this.course$ = this.db.getQuestion(this.qId).pipe(flatMap(q => this.db.getCourse(q.course)));
+    this.userId = this.auth.currentUserId;
+    this.solvedQuestions$ = this.db.getSolvedQuestions(this.userId);
+    this.solvedQuestions$.subscribe(value => {
```

Comment:
This is bad. You query for all the questions the user solved just to check if he solved a single question? Check it directly without going over all the questions..

Reply:
And make it observable, do not subscribe in the code but use async pipe.

Reply:
> And make it observable, do not subscribe in the code but use async pipe.

The async pipe is used in the HTML file, right?

Reply:
What it the problem with subscribing?